### ISSUE TITLE
Implement SinonJS to improve unit test readability

### ISSUE THREAD

**Issue by JohnnyEstilles** [pinoyyid/ngGAPI#2]
I believe the project's unit tests could benefit greatly form the implementation of a spying/stubbing/mocking library such as [SinonJS](http://sinonjs.org/).

For example, a test such as:
````
js
it('insert media should fail for invalid params or data', function () {
  ...
  var ro = DriveService.files.insert({title: 'title-'+id}, {uploadType:'resumable'}, 'notb64');
  ro.promise.then(
    function () {expect('should have failed D136 no resumable yet').toBe('false')},
    function (reason) {expect(reason).toMatch('D136')}
  );
  ...
});
```
could be written as:
```js
it('insert media should fail for invalid params or data', function () {
  var success = sinon.stub();
  var failure = sinon.stub();
  ...
  var ro = DriveService.files.insert({title: 'title-'+id}, {uploadType:'resumable'}, 'notb64');
  ro.promise
    .then(success, failure)
    .finally(function() {
      expect(success).not.toHaveBeenCalled();
      expect(failure).toHaveBeenCalledWith('D136')
    });
  ...
});
```

I can submit a PR implementing Sinon and updating all the existing unit tests, if you're open to the idea.

**Comment by pinoyyid** [comment]
Yes, I agree that's far more elegant.

**Comment by JohnnyEstilles** [comment]
I have a PR ready for this. Will submit as soon as PR #4 is merged.

**Comment by JohnnyEstilles** [comment]
Implemented via PR #5.
